Plugins~/Src/MeshSyncClientBlender/msblenContext.cpp

```diff
@@ -740,9 +740,12 @@ ms::MeshPtr msblenContext::exportMesh(msblenContextState& state, msblenContextPa
                 Depsgraph* depsgraph = bl::BlenderPyContext::get().evaluated_depsgraph_get();
                 bobj = (Object*)bl::BlenderPyID(bobj).evaluated_get(depsgraph);
             }
-            if (Mesh *tmp = bobj.to_mesh()) {
+            if (Mesh* tmp = bobj.to_mesh()) {
                 data = tmp;
-                m_meshes_to_clear.push_back(src);
+                // Only clear meshes that aren't created from others:
+                if (src->id.orig_id == NULL) {
```

Comment:
Is this related to this particular fix/feat ?

Reply:
I think it's related, it didn't happen until we tried to instance lights.

Reply:
Other meshes. The issues is that blender internally creates temporary meshes that seem to be copies of other meshes and frees them later. When meshsync tries to access those later, the stored pointer points to freed memory and we get exceptions.